packages/api/src/__tests__/e2e/mapi/mapi.test.e2e.ts

```diff
@@ -274,36 +346,157 @@ describe("MAPI E2E Tests", () => {
+    expect(initConversionProgress?.status).toEqual("completed");
   });
 
+  it("received WH with correct meta", async () => {
+    const whRequest = getConsolidatedWebhookRequest();
+    expect(whRequest).toBeTruthy();
+    if (!whRequest) throw new Error("Missing WH request");
+    expect(whRequest.meta).toBeTruthy();
+    expect(whRequest.meta).toEqual(
+      expect.objectContaining({
+        type: "medical.consolidated-data",
+        messageId: expect.anything(),
+        when: expect.anything(),
+      })
+    );
+    expect(validateUuid(whRequest.meta.messageId)).toBeTrue();
+    const minDate = dayjs().subtract(1, "minute").toDate();
+    const maxDate = dayjs().add(1, "minute").toDate();
+    expect(dayjs(whRequest.meta.when).isBetween(minDate, maxDate)).toBeTrue();
+    expect(whRequest.meta).not.toEqual(
+      expect.objectContaining({
+        data: expect.toBeFalsy,
+      })
+    );
+  });
+
+  it("received WH with MR in PDF format", async () => {
+    const whRequest = getConsolidatedWebhookRequest();
+    const consolidatedData = whRequest?.patients;
+    expect(consolidatedData).toBeTruthy();
+    if (!consolidatedData) throw new Error("Missing consolidated data");
+    expect(consolidatedData.length).toEqual(1);
+    expect(consolidatedData[0].status).toEqual("completed");
+    const bundle = consolidatedData[0].bundle;
+    expect(bundle).toBeTruthy();
+    if (!bundle) throw new Error("Missing Bundle");
+    expect(bundle.type).toEqual("searchset");
+    expect(bundle.resourceType).toEqual("Bundle");
+    expect(bundle.total).toEqual(1);
+    expect(bundle.entry).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({
+          resource: expect.objectContaining({
+            resourceType: "DocumentReference",
+            content: expect.arrayContaining([
+              expect.objectContaining({
+                attachment: expect.objectContaining({
+                  contentType: "application/pdf",
+                }),
+              }),
+            ]),
+          }),
+        }),
+      ])
+    );
+    const resource = bundle.entry?.[0]?.resource;
+    if (!resource || resource.resourceType !== "DocumentReference") {
+      throw new Error("Missing DocumentReference");
+    }
+    const patientRef = resource.subject?.reference;
+    expect(patientRef).toBeTruthy();
+    expect(patientRef).toEqual(`Patient/${patient.id}`);
+    url = resource.content?.[0].attachment?.url;
+    expect(url).toBeTruthy();
+    expect(isValidUrl(url)).toEqual(true);
+  });
+
+  test("can download MR", async () => {
+    if (!url) return;
+    const fileBuffer = await downloadToMemory({ url });
+    expect(fileBuffer).toBeTruthy();
+    mrContentBuffer = fileBuffer;
+  });
+
+  test("MR in PDF format has expected contents", async () => {
+    if (!mrContentBuffer) return;
+    const fileType = detectFileType(mrContentBuffer);
+    expect(fileType).toBeTruthy();
+    expect(fileType.mimeType).toEqual(PDF_MIME_TYPE);
```

Comment:
for the html version of this, we compare the contents to a template html. Do we want to do the same for a template pdf?

Reply:
The file is dynamically generated, so it would be much harder to get a PDF dynamically generated to be matched against the result. Considering its just rendering of the same HTML file, I think its fine as is for the time being.